### ISSUE TITLE
ROW Permit  addendum Updates and TURP & DS permit bug: Formatting issues and QR Code (SS) (On PROD)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#12031]
### Various updates to include:
1. Adding QR code to addendum
2. Updating various formatting issues on addendum
3. Debugging a formatting issue happening on TURP and DS. (Perhaps others as well.)
     - 50511 (TURP), 50506 (DS)

----
![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/a1bd3672-c647-4a7d-a6e2-7c4f029b07d8)

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/5f24eadb-effc-4d0b-a3c3-ad0183cf0fe9)

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/03b83c02-7d7f-45b5-a2ec-b7683ad97887)

> What application are you using?

AMANDA

> Describe the problem.

The following edits are requested for the permit addendum that is present on the final page of the permit report on all ROW permit folder types (DS, EX, RW and PSM):

Under the current 
- line item #2 ("_All traffic control must be..._"), the text that reads "_Transportation Criterial Manual (TCM)_" should be corrected to "`Transportation Criteria Manual (TCM)`", with the word "Criterial" corrected to "Criteria". 
- The hyperlink should also be provided over this entire text, as right now it misses the first `T` in "`Transportation`".

- An additional line item should be added after the 
current line item # 3 that reads as follows: "`Prior to any excavation work in the ROW, underground utilities must be located and marked at the surface using the Uniform Color Codes of the American Public Works Association (APWA).`"

> How soon do you need this?

Flexible  — An extended timeline is OK

> Requested By
Kyle K.

Permit Addendum links:

[https://app.zenhub.com/files/140626918/ea3e9bb4-12b7-404d-ba7c-9e792bbfa64f/download](https://app.zenhub.com/files/140626918/ea3e9bb4-12b7-404d-ba7c-9e792bbfa64f/download)

Request ID: DTS23-107160

**Comment by SubhaShroff** [comment]
Notes: Following are the identified reports that need the addendum update as per AC -
DS - 50506 
RW-TURP - 50511
RW - ROW - 50510
RW - Film - 50509
RW - Parking Permit - 50512
RW - Street Event  - 50513
RW - Street Vendor - 50503
RW - Valet Parking Annual - 50504
RW - Valet Operator - 50505
RW - Temporary Valet Parking - 50502

EX - 50510
PSM - 85008

**Comment by Nadin-Nader** [comment]
Subha found that the links were not doing anything. Reached out to Kyle and he was able to provide the links for us i the word doc attached[https://app.zenhub.com/files/140626918/e68e124a-f7e3-435b-bd41-19183c1c5e7f/download](https://app.zenhub.com/files/140626918/e68e124a-f7e3-435b-bd41-19183c1c5e7f/download)

**Comment by SubhaShroff** [comment]
Thanks @Nadin-Nader  It is the same above document copy you shared with me. As I can't access the link, I just uploaded the document here for my reference. [Permit Addendum Links.docx](https://github.com/cityofaustin/atd-data-tech/files/11423595/Permit.Addendum.Links.docx)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I updated the addendum for TURP permit, the sample output pdf is attached here for review. I will be updating the same in other reports as well. As I thought it can't be copy paste as I have to create formula field for each link on each report, just fyi.
[50511_f_Temp_Use_Permit_ROW_Use.pdf](https://github.com/cityofaustin/atd-data-tech/files/11478800/50511_f_Temp_Use_Permit_ROW_Use.pdf)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The above sample I have attached is taken directly from Crystal Report tool and the links are working in it. After uploading in Amanda and the report generated from Amanda, the URLs are not working. That's weird. I will escalate this to Jenice. Probably, she can take it to Granicus.

Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  this morning, I had joined a tech call with Jenice and other CTM team to discuss this issue that the URLs not working on Amanda generated report before we take it Granicus. After looking into the issue, Jenice suggested us to take this to Granicus. I will be joining them tomorrow for this isse. THanks.

**Comment by Nadin-Nader** [comment]
update:
as far as original AC, the text is changed and its deployed in TEST. But, later i found the URLs are already not working. Yesterday, I joined a call with Jenice and other CTM team for this issue. After discussion, Jenice suggested we take this to Granicus and the call is this evening 4pm. Hopefully, Granicus should let us know whats wrong with the URLs and can provide a fix.
Meeting should be today at 4pm

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Just spoke to Granicus team, they asked to raise a ticket and they will investigate. Jenice is going to raise a ticket and keep us updated. Since, its with Granicus, we can expect a time elapse on this. Thanks.

**Comment by Nadin-Nader** [comment]
@SubhaShroff here is what Kyle said:

[1:16 PM] Kampe, Kyle
Thanks Nate. It does look a little wonkier than I think we'd be okay with. The extra spaces in between the hyperlinked text and non-hyperlinked text aren't a big deal, but what I'm more concerned about is how some of the hyperlinks are pretty far off their line. It just looks a little to shoddy for what we want on the official permit report.

**Comment by TracyLinder** [comment]
### Update w/ ROW

- The hyperlinks were not working, and when fixing it messes up the formatting. 
- Subha has tried to fix with Sree & Priya - no fixes
- Subha escalated to Granicus (AMANDA Vendor) for help
- Will place this in Ongoing until Graniucs responds.

**Comment by KyleErich** [comment]
@Nadin-Nader, Can we add the following additional change to the permit addendum along with this card? I can add to the AC if so.
![Capture](https://github.com/cityofaustin/atd-data-tech/assets/130776506/638e6f4d-e49e-47cb-8e7b-63e1076b1407)

**Comment by TracyLinder** [comment]
### Friday ROW Review
- ROW asked for an update so they can update their execs.
- Tracy will ask Jenice for a timeframe and which release this fix is in.

**Comment by TracyLinder** [comment]
### Update with Jenice 8/22/23

- Jenice Remembers giving us a work-around, remembers talking to Granicus
- Jenice linked us to Granicus video that shows a work around
- Granicus is working on a permanent fix.

**Comment by TracyLinder** [comment]
@SubhaShroff please watch this video provided by Granicus as a work-around for this issue. 
https://www.screencast.com/t/XHofgzKkwK

**Comment by TracyLinder** [comment]
FYI we have a new ticket to update the addendum, #12870

**Comment by TracyLinder** [comment]
### Friday ROW Review 9/15/23

Per Subha the video provided above by Granicus. is the same technique she already tried and it **didn't work**. 


Subha believes this is an issue with AMANDA, not with crystal reports. This works when she views it with Crystal Reports. 

Nate suggests rethinking how we're formatting this issue, so we can work around this issue versus waiting on Granicus to help us fix the root cause.

Tracy suggested exploring using a lookup table as an option, and Subha will explore that option. 

**The 3 Solutions we can choose from:**

**Comment by TracyLinder** [comment]
### Approved QR Template

![MicrosoftTeams-image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/62842915-9432-49c1-b829-be02cf05d542)

**Comment by SubhaShroff** [comment]
@Nadin-Nader @TracyLinder I tried some random existing URL link with other text to try URL links to work. Here is the download of the Amanda output. The 4th page has the text I tried. You can see the highlight link still not working. 
The text I tried queried using this query: SELECT lookupstring3 FROM validlookup WHERE lookupcode=83915 AND lookup1=47;
[Film Closure Permit ROW(50509)- Trial with LOOKUP.pdf](https://github.com/cityofaustin/atd-data-tech/files/12652485/Film.Closure.Permit.ROW.50509.-.Trial.with.LOOKUP.pdf)

### This is the downloaded pdf file of output from Crystal Report Tool. We can see the (4th page) the links are working.
[50509_f_Film_Closure_Permit_ROW.pdf](https://github.com/cityofaustin/atd-data-tech/files/12652586/50509_f_Film_Closure_Permit_ROW.pdf)

Now going to put the embedded textboxes for the links which works but aligning the each text box with the rest of the text is challenging as we already saw/spoke about.
Work in Progress.....

**Comment by SubhaShroff** [comment]
50506 DS Permit -

- Overlapping data is fixed. **But for that the format layout is changed a bit in the first page for this report to fix the issue.** Please check if that will work.
- Addendum is added - tried my best to fit those aligned.
- QR code is added.

[Driveway SideWalk Permit(50506)- Result.pdf](https://github.com/cityofaustin/atd-data-tech/files/12677409/Driveway.SideWalk.Permit.50506.-.Result.pdf)

Working on other reports...........................

**Comment by SubhaShroff** [comment]
@Nadin-Nader @KyleErich  50505 - The original report - Valet Operator Permit (50505) report doesn't have addendum. Do you want me to add to it along with QR code?

PSM report orientation is of landscape. So, the Permittee text and QR code won't fit in the same page of addendum. Those things move to next page. I have attached the output report here. Please review.
[85008_PSM_Permit_Report.pdf](https://github.com/cityofaustin/atd-data-tech/files/12783446/85008_PSM_Permit_Report.pdf)

And also, I sent @Nadin-Nader  the reports  except 50505, in email to copy in DEV/TEST server. Thank you.

**Comment by SubhaShroff** [comment]
Thank you @KyleErich  for your kind and quick responses. I have all the reports on DEV/TEST and tested on AMANDA TEST. I will test for portal reports also for the same list after copying these reports on respective folders on DEV server.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich  As I debugged I realized what shows up on CC is logged in AMANDA user email id. In the below screenshot, it shows my email id. Since Kyle was logged in to test it shows Kyle's email id and applicant also has the same email id. I also, tested it for multiple permits and confirm that it doesn't add duplicate emails.
Even as if it adds same email-id on TO, CC, BCC the system still sends only one email but not multiples.

**Comment by Nadin-Nader** [comment]
This has been sent to the CTM Team to be deployed to the C drive on DEV and TEST under **SCTASK0360680**

**Comment by Nadin-Nader** [comment]
@SubhaShroff This has been sent to CTM to be deployed to DEV and TEST Portal servers under **SCTASK0361855**

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader. 
UPDATE: Jenice deployed the reports for DEV/TEST portal. I retested all on TEST AMANDA and Portal. 50506, 50511, 85008 are not updated. So I have sent an email to Jenice to redeploy the reports on DEV/TEST portals.
Waiting to hear from her. The rest of reports are good.

**Comment by SubhaShroff** [comment]
@Nadin-Nader 50506 DS Permit report is still hasn't updated on TEST portal, surprisingly. I have set up a call with Jenice tomorrow morning to have a look at it. Probably a day change also set thing right.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  This morning had a call with Jenice and you to fix 50506 on TEST portal. 
It is fixed and tested all the reports for addendum and QR code. All look good on AMANDA as well the Portal. 
But it is noticed that currently(already existing issue), the barcode doesn't show up on TEST/PROD Portal reports, instead it displays the folderrsn. As we discussed it needs a probable TEST Portal server refresh after Jenice copied the barcode font on portal servers to fix that and you are reaching her for server bounce.
![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/4a6275a3-eb6a-4d22-bf9d-424970003971)

### **PORTAL:**
![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/8adcae81-02be-4ed5-aa24-0556309a43af)

NOTE for future ref: The problem with 50506 report was on TEST portal it is configured to be referred as 50506_f_Driveway_Side_Walk_Permit_prod.rpt file. But we were uploading it with 50506_f_Driveway_Side_Walk_Permit.rpt. Name change fixed it.

**Comment by SubhaShroff** [comment]
@Nadin-Nader As far as the AC all the reports are set, verified on TEST AMANDA and Portal also. Regarding barcode on the first page thats not displaying on TEST/PROD portals, I think, we should take it as a new ticket. There is nothing I should change on reports for that. CTM team has to set the font on those servers. 
So, as per me its ready for UAT on TEST.

**Comment by KyleErich** [comment]
@SubhaShroff @Nadin-Nader UAT on TEST approved.

**Comment by SubhaShroff** [comment]
Thanks @Kyle.
@Nadin-Nader  PROD deployment request SCTASK0364282 is raised for this ticket.

**Comment by Nadin-Nader** [comment]
Updated the priority and sent Deanna a message on Teams asking her to deploy the reports as soon as she could. Waiting on Deployment or response.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to PROD

**Comment by Nadin-Nader** [comment]
Waiting on UAT response from Kyle

**Comment by Nadin-Nader** [comment]
Kyle has approved this on PROD. Closing.
